socs/agents/scpi_psu/agent.py

```diff
@@ -41,14 +44,37 @@ def init(self, session, params=None):
             if not acquired:
                 return False, "Could not acquire lock"
 
-            try:
-                self.psu = PsuInterface(self.ip_address, self.gpib_slot)
-                self.idn = self.psu.identify()
-            except socket.timeout as e:
-                self.log.error(f"PSU timed out during connect: {e}")
-                return False, "Timeout"
+            if self.port is None:  # Use the old Prologix-based GPIB code
+                try:
+                    self.psu = PsuInterface(self.ip_address, self.gpib_slot)
+                    self.idn = self.psu.identify()
+                except socket.timeout as e:
+                    self.log.error(f"PSU timed out during connect: {e}")
+                    return False, "Timeout"
+            else:  # Use the new direct ethernet connection code
+                try:
+                    self.psu = ScpiPsuInterface(self.ip_address, self.gpib_slot, port=self.port)
+                    self.idn = self.psu.identify()
+                except socket.timeout as e:
+                    self.log.error(f"PSU timed out during connect: {e}")
+                    return False, "Timeout"
+                except ValueError as e:
+                    if (e.args[0].startswith('Model number')):
+                        self.log.error(f"PSU initialization error: {e}. Suggest appending {e.args[-1]} to the list of known model numbers in scpi_psu/drivers.py")
```

Comment:
This is a good suggestion, but can you wrap the line to <100 characters?

Reply:
Fixed